partner_identification_stock/models/stock_picking.py

```diff
@@ -0,0 +1,43 @@
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _compute_id_requirement(self):
+        for rec in self:
+            rec.require_id = rec.valid_id = False
+            for line in rec.move_lines:
+                if line.product_id.require_id_stock_ids:
+                    rec.require_id = True
+                ffl_lines = rec.partner_id.id_numbers.filtered(
+                    lambda i: i.category_id.id
+                    in line.product_id.require_id_stock_ids.ids
+                    and i.valid_until
+                )
+                if any(
+                    line.valid_until >= fields.Date.today()
+                    and line.status in ["open", "pending"]
+                    for line in ffl_lines
+                ):
+                    rec.valid_id = True
+
+    require_id = fields.Boolean(string="Require ID?", compute="_compute_id_requirement")
+    valid_id = fields.Boolean(string="Valid ID?", compute="_compute_id_requirement")
+
+    def action_confirm(self):
```

Comment:
As you require valid partner identification for 'confirmation', this should appears in module naming.

So partner_identification_stock_picking_confirmation is longer but better.

Reply:
We should not change the module name, as it might affect other modules (if any module depends on this one).

Reply:
> We should not change the module name, as it might affect other modules (if any module depends on this one).

This module is not yet merged. So, this can still be done